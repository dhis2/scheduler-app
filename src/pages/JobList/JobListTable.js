import React from 'react'
import { object, arrayOf, string } from '@dhis2/prop-types'
import {
    Table,
    TableHead,
    TableRowHead,
    TableRow,
    TableCell,
    TableCellHead,
    TableBody,
} from '@dhis2/ui-core'
import i18n from '@dhis2/d2-i18n'
import JobListTableItem from './JobListTableItem'

const JobListTable = ({ jobIds, jobEntities }) => (
    <Table>
        <TableHead>
            <TableRowHead>
                <TableCellHead>{i18n.t('Job name')}</TableCellHead>
                <TableCellHead>{i18n.t('Type')}</TableCellHead>
                <TableCellHead>{i18n.t('Schedule')}</TableCellHead>
                <TableCellHead>{i18n.t('Next run')}</TableCellHead>
                <TableCellHead>{i18n.t('Status')}</TableCellHead>
                <TableCellHead>{i18n.t('On/off')}</TableCellHead>
                <TableCellHead />
            </TableRowHead>
        </TableHead>
        <TableBody>
            {jobIds.length === 0 ? (
                <TableRow>
                    <TableCell>{i18n.t('No jobs to display')}</TableCell>
                </TableRow>
            ) : (
                jobIds.map(id => (
                    <JobListTableItem key={id} job={jobEntities[id]} />
                ))
            )}
        </TableBody>
    </Table>
)

JobListTable.propTypes = {
    jobEntities: object.isRequired,
    jobIds: arrayOf(string).isRequired,
}

export default JobListTable
