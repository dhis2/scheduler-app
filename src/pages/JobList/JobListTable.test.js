import React from 'react'
import { shallow } from 'enzyme'
import JobListTable from './JobListTable'

describe('<JobListTable>', () => {
    it('renders without errors when there are jobs', () => {
        const jobIds = ['1', '2']
        const jobEntities = {
            1: {
                cronExpression: '0 0 * ? * *',
                displayName: 'Name',
                enabled: true,
                id: '1',
                jobStatus: 'ENABLED',
                jobType: 'Type',
                nextExecutionTime: '2100-10-10T14:48:00',
                schedulingType: 'CRON',
            },
            2: {
                delay: 6000,
                displayName: 'Name',
                enabled: true,
                id: '2',
                jobStatus: 'ENABLED',
                jobType: 'Type',
                nextExecutionTime: '',
                schedulingType: 'FIXED_DELAY',
            },
        }

        shallow(<JobListTable jobIds={jobIds} jobEntities={jobEntities} />)
    })

    it('renders without errors when there are no jobs', () => {
        const jobIds = []
        const jobEntities = {}

        shallow(<JobListTable jobIds={jobIds} jobEntities={jobEntities} />)
    })
})
