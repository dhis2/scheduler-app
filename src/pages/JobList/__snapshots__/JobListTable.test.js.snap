// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobListTable> renders correctly when there are jobs 1`] = `
<Table
  dataTest="dhis2-uicore-table"
>
  <TableHead
    dataTest="dhis2-uicore-tablehead"
  >
    <TableRowHead
      dataTest="dhis2-uicore-tablerowhead"
    >
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Job name
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Type
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Schedule
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Next run
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Status
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        On/off
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      />
    </TableRowHead>
  </TableHead>
  <TableBody
    dataTest="dhis2-uicore-tablebody"
  >
    <JobListTableItem
      job={
        Object {
          "cronExpression": "",
          "displayName": "",
          "enabled": true,
          "id": "1",
          "jobStatus": "",
          "jobType": "",
          "nextExecutionTime": "",
        }
      }
      key="1"
    />
  </TableBody>
</Table>
`;

exports[`<JobListTable> renders correctly when there are no jobs 1`] = `
<Table
  dataTest="dhis2-uicore-table"
>
  <TableHead
    dataTest="dhis2-uicore-tablehead"
  >
    <TableRowHead
      dataTest="dhis2-uicore-tablerowhead"
    >
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Job name
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Type
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Schedule
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Next run
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        Status
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      >
        On/off
      </TableCellHead>
      <TableCellHead
        dataTest="dhis2-uicore-tablecellhead"
      />
    </TableRowHead>
  </TableHead>
  <TableBody
    dataTest="dhis2-uicore-tablebody"
  >
    <TableRow
      dataTest="dhis2-uicore-tablerow"
    >
      <TableCell
        dataTest="dhis2-uicore-tablecell"
      >
        No jobs to display
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;
